syntax = "proto3";

package ledger;

service LedgerService {
  rpc ProcessReceipt (ReceiptRequest) returns (ReceiptResponse);
  rpc ProcessReceipts (ReceiptsRequest) returns (ReceiptsResponse);
}

message ReceiptRequest {
  bytes image_data = 1;
}

message ReceiptResponse {
  string json_output = 1;
}

message ReceiptsRequest {
  repeated bytes image_data = 1;
}

message ReceiptsResponse {
  string json_output = 1;
}
